pkg/machines/components/vcpuModal.jsx

```diff
@@ -0,0 +1,291 @@
+import React from 'react';
+import cockpit from 'cockpit';
+
+import { show_modal_dialog } from 'cockpit-components-dialog.jsx';
+import SelectComponent from 'cockpit-components-select.jsx';
+import InfoRecord from './infoRecord.jsx';
+import { Alert } from './notification/inlineNotification.jsx';
+import { setVCPUSettings } from "../actions.es6";
+
+React;
+
+const _ = cockpit.gettext;
+
+const dividers = (num) => {
+    const divs = [1];
+
+    for (let i = 2; i < num; i++) {
+        if (num % i === 0) {
+            divs.push(i);
+        }
+    }
+
+    if (num > 1) {
+        divs.push(num);
+    }
+
+    return divs;
+}
+
+const clamp = (value, max, min) => {
+    return value < min || isNaN(value) ? min : (value > max ? max : value) ;
+}
+
+const Select = function ({ id, items, onChange, value }) {
+    return ( <SelectComponent.Select id={id} initial={value} onChange={onChange}>
+        {items.map((t) => (
+            <SelectComponent.SelectEntry data={t}>{t}</SelectComponent.SelectEntry>
+        ))}
+    </SelectComponent.Select>);
+}
+
+class VCPUModalBody extends React.Component {
+    constructor(props) {
+        super(props);
+        let state = {
+            sockets: props.cpu.topology.sockets || 1,
+            threads: props.cpu.topology.threads || 1,
+            cores: props.cpu.topology.cores || 1
+        }
+        if (!props.isOvirt) {
```

Comment:
This is the very first occurrence of `oVirt` notion within `pkg/machines` and I would like to avoid it.
There's concept of `provider`, please reuse it in a generic way.